### PR TITLE
Move the semantic conventions generated code into its own module

### LINKED ISSUE #2264: Extract the semantic conventions classes into their own module.
The [OTEP for versioning](https://github.com/open-telemetry/oteps/pull/143) strongly suggests that semantic conventions should be considered a separate module. They will probably evolve more quickly than other parts of the API, and it makes sense to be able to consume them independently.

### PR DESCRIPTION
**Pull request by jkwatson**
Since the semantic conventions may be versioned independently of the rest of the API, it seems like a good idea to move them to their own module before we hit 1.0. 

Resolves #2264

### CODE REVIEW

File: sdk/trace/build.gradle [hunk 1]
```diff
@@ -12,6 +12,7 @@ ext.propertiesDir = "build/generated/properties/io/opentelemetry/sdk/trace"
 
 dependencies {
     api project(':api:all'),
+            project(':api:semconv'),
```
Comment (bogdandrutu) [inline sdk/trace/build.gradle:15]: maybe just `semconv`? Not under `api`?

**Review by bogdandrutu — APPROVED** [review]

File: sdk/trace/build.gradle (hunk 1 above)
Comment (jkwatson) [reply sdk/trace/build.gradle:15]: I wasn't really sure which was better. Looking at the specs, it looks like semantic conventions are described under a section called "data", so I agree that it probably doesn't make sense to have it under API here, either.  👍

**Review by Oberon00 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -16,12 +16,12 @@ project.
 This project contains the following top level components:
 
 * [OpenTelemetry API](api/):
-  * [tracing api](api/all/src/main/java/io/opentelemetry/api/trace/) includes `Tracer`, `Span` and `SpanContext`.
-  * [baggage api](api/all/src/main/java/io/opentelemetry/api/baggage/) defines a collection of entries in the form of key-value pairs of data that can be propagated to provide contextual information.
-  * [metrics api](api/metrics/src/main/java/io/opentelemetry/api/metrics/).
-  * [context api](api/context/src/main/java/io/opentelemetry/context/).
+  * [stable apis](api/all/src/main/java/io/opentelemetry/api/all/) including `Tracer`, `Span`, `SpanContext`, and `Baggage`
+  * [semantic conventions](api/semconv/) Generated code for the OpenTelemetry semantic conventions.
```
Comment (anuraaga) [inline README.md:20]:
```suggestion
  * [semantic conventions](semconv/) Generated code for the OpenTelemetry semantic conventions.
```

**Review by anuraaga — APPROVED** [review]
I'm hoping we won't have to version it differently since it seems like it could confusing for one artifact to have its own version (otherwise, we would wonder why we don't do the same for other modules like `context).

But a separate artifact makes sense anyways since we can expect backends to just need proto and semconv, and not API.

**Comment by jkwatson** [comment]
> I'm hoping we won't have to version it differently since it seems like it could confusing for one artifact to have its own version (otherwise, we would wonder why we don't do the same for other modules like `context).


Oh for sure. I really don't want to, but if we need to, it would only be possible if it's broken out. :)

**Review by bogdandrutu — APPROVED** [review]
Should the package be `io.opentelemetry.semconv`? Without `api`?

**Comment by jkwatson** [comment]
> Should the package be `io.opentelemetry.semconv`? Without `api`?

oh yes. that does make sense. good eye! I'll regenerate the file.

**Comment by jkwatson** [comment]
ok, I have repackaged the `SemanticAttributes` into `io.opentelemetry.semconv`, and deprecated the ones in the api package. Thanks for the reviews!

File: semconv/build.gradle [hunk 3]
```diff
@@ -0,0 +1,16 @@
+plugins {
+    id "java-library"
+    id "maven-publish"
+
+    id "me.champeau.gradle.jmh"
```
Comment (anuraaga) [inline semconv/build.gradle:5]:
```suggestion
```

File: semconv/build.gradle [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    id "ru.vyarus.animalsniffer"
+}
+
+description = 'OpenTelemetry Semantic Conventions'
+ext.moduleName = "io.opentelemetry.api.semconv"
```
Comment (anuraaga) [inline semconv/build.gradle:10]:
```suggestion
ext.moduleName = "io.opentelemetry.semconv"
```

**Review by anuraaga — APPROVED** [review]
Found just small nits

**Pull request merged**